pkg/csi/service/vanilla/controller.go

```diff
@@ -1624,23 +1637,32 @@ func (c *controller) ListVolumes(ctx context.Context, req *csi.ListVolumesReques
 }
 
 func (c *controller) processQueryResultsListVolumes(ctx context.Context, startingToken int, endIndex int,
-	queryResult *cnstypes.CnsQueryResult, allNodeVMs []*cnsvsphere.VirtualMachine) ([]*csi.ListVolumesResponse_Entry,
+	cnsVolumes []cnstypes.CnsVolume, allNodeVMs []*cnsvsphere.VirtualMachine) ([]*csi.ListVolumesResponse_Entry,
 	string, string, error) {
 
 	volumeType := ""
 	nextToken := ""
 	log := logger.GetLogger(ctx)
 	var entries []*csi.ListVolumesResponse_Entry
 
-	volumeIDToNodeUUIDMap, err := getBlockVolumeToHostMap(ctx, c.manager, allNodeVMs)
+	volumeIDToNodeUUIDMap, err := getBlockVolumeToHostMap(ctx, c, allNodeVMs)
 	if err != nil {
 		return entries, nextToken, volumeType, err
 	}
 
 	for i := startingToken; i < endIndex; i++ {
-		if queryResult.Volumes[i].VolumeType == common.FileVolumeType {
+		if cnsVolumes[i].VolumeType == common.FileVolumeType {
+			// If this is multi-VC configuration, then
+			// skip processing query results for file volumes
+			if multivCenterCSITopologyEnabled {
+				logger.LogNewErrorCodef(log, codes.Internal,
```

Comment:
Instead of returning an error. Can we just print a log message and return a ListVolume response for block volumes?

Reply:
changed this to debug log message